src/Starcounter.Database.Extensions/TransactorBase.cs

```diff
@@ -125,6 +163,19 @@ protected virtual async Task<T> ExecuteCallback<T>(IDatabaseContext db, Func<IDa
         /// <param name="db">The database context returned by EnterContext.</param>
         /// <param name="exceptionThrown">True if an exception was thrown when invoking the
         /// delegate; false otherwise.</param>
-        protected virtual void LeaveContext(IDatabaseContext db, bool exceptionThrown) { }
+        protected virtual void LeaveContext(TTransactorContext transactorContext, IDatabaseContext db, bool exceptionThrown) { }
+
+        /// <summary>
+        /// Invoked right after a transaction commit and outside of its scope.
+        /// This method is not invoked if the transaction fails.
+        /// </summary>
+        protected virtual void LeftContext(TTransactorContext transactorContext) { }
+
+        /// <summary>
+        /// Creates a <see cref="TTransactorContext"/> instance required for <see cref="EnterContext(TTransactorContext, IDatabaseContext)"/> and <see cref="LeaveContext(TTransactorContext, IDatabaseContext, bool)"/> methods.
+        /// Returns null by default.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual TTransactorContext CreateTransactorContext() => null;
```

Comment:
Can we name them so that they clearly go in pair: `CreateTransactorContext/DestroyTransactorContext` or whatever?